src/dotnet/APIView/docs/apiview-lightweight-permissions-proposal.md

```diff
@@ -0,0 +1,277 @@
+
+| Role | Description |
+|------|-------------|
+| **ServiceTeam** | Service team members who create and manage reviews |
+| **SdkTeam** | SDK team members with additional privileges |
+| **Admin** | Highest level permissions, for APIView maintainers only |
+
+### Language-Scoped Roles
+These roles **must** be assigned to a specific language:
+
+| Role | Description |
+|------|-------------|
+| **DeputyArchitect** | Similar permissions to Architect, for designated backups |
+| **Architect** | Elevated permissions for API review approval |
+
+---
+
+## 3. Permission Matrix
+
+| Action | ServiceTeam | SdkTeam | DeputyArchitect | Architect | Admin |
+|--------|:-----------:|:-------:|:---------------:|:---------:|:-----:|
+| View reviews | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Create reviews | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Add comments | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Delete own comments | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Delete any comment | | | ✓ | ✓ | ✓ |
+| Delete all copilot comments | | | | | ✓ |
+| Approve API revision | | | ✓* | ✓* | ✓ |
+| Approve namespace | | | ✓* | ✓* | ✓ |
+| Delete revision | ✓ | ✓ | ✓ | ✓ | ✓ |
+| Delete entire review | | | | | ✓ |
+| Manage user permissions | | | | | ✓ |
+| Access admin features | | | | | ✓ |
+
+*\* Language-scoped: Only for languages where user has this role*
+
+---
+
+## 4. Data Storage
+
+Create a new `Permissions` container in Cosmos DB.
+
+### Role Assignment Schema (Discriminated Union)
+
+Role assignments use a discriminated union pattern with a `kind` field to distinguish between global and language-scoped roles. This design **enforces at the schema level** that:
+- Global roles (`Admin`, `SdkTeam`, `ServiceTeam`) cannot have a language
+- Language-scoped roles (`Architect`, `DeputyArchitect`) must have a language
+
+**Global Role Assignment:**
+```json
+{ "kind": "global", "role": "SdkTeam" }
+```
+
+**Language-Scoped Role Assignment:**
+```json
+{ "kind": "scoped", "role": "Architect", "language": "Python" }
+```
+
+### Group Permission Document
+```json
+{
+  "id": "group-python-architects",
+  "type": "group",
+  "groupId": "python-architects",
+  "groupName": "Python Architects",
+  "roles": [
+    { "kind": "scoped", "role": "Architect", "language": "Python" }
+  ],
+  "members": ["almend", "johanste"],
+  "lastUpdatedOn": "2024-12-04T00:00:00Z",
+  "lastUpdatedBy": "admin"
+}
+```
+---
+
+## 5. Effective Permissions Resolution
+
+The key to this system is computing a user's **effective permissions** by merging their direct role assignments with roles inherited from group memberships.
+
+```csharp
+public async Task<EffectivePermissions> GetEffectivePermissionsAsync(string userId)
+{
+    // 1. Get all groups where user is a member
+    var groups = await _permissionRepo.GetGroupsForUserAsync(userId);
+    
+    // 2. Merge group-inherited, highest role wins per language
+    return MergePermissions(groups);
+}
+```
+
+**Example**: User has direct `SdkTeam` (global) + group membership gives `Architect` (Python)
```

Comment:
Aren't all memberships via groups now? So the user would be in an SdkTeam group and a Python architects group?